src/videojs-hls.js

```diff
@@ -950,6 +954,7 @@ videojs.Hls.prototype.drainBuffer = function(event) {
   // the playlist
   if (this.duration() !== Infinity && mediaIndex + 1 === playlist.segments.length) {
     this.mediaSource.endOfStream();
+    this.lastSeekedTime_ = undefined;
```

Comment:
So that the condition above returns true, and we know to reset. It's not wiped out automatically at the end of playback, but continues to store the last seeked value (if the user ever seeked at all).